vespa/deployment.py

```diff
@@ -550,127 +646,185 @@ def deploy(
+        )
 
-        # We need to wait for the tests to finish before we can monitor the deployment itself
-        self._follow_deployment(instance, "staging-test", run_id)
-        self._follow_deployment(instance, "system-test", run_id)
+    def get_application(
+        self, instance: str = "default", environment: str = "dev", max_wait: int = 60
+    ) -> Vespa:
+        """
+        Get a connection to the Vespa application instance.
+        Will only work if the application is already deployed.
 
-        # Like with the run id, it can take a couple of seconds for the job to show up here.
-        # TODO Replace with a more robust solution
-        sleep(20)
-        region = self.get_prod_region()
-        self._follow_deployment(instance, f"production-{region}", run_id)
+        Example usage::
+
+            vespa_cloud = VespaCloud(...)
+            app: Vespa = vespa_cloud.get_application()
+            # Feed, query, visit, etc.
+
+        :param instance: Name of this instance of the application, in the Vespa Cloud. Default is "default".
+        :param environment: Environment of the application. Default is "dev". Options are "dev" or "prod".
+        :param max_wait: Seconds to wait for the application to be up. Default is 60 seconds.
+
+        :return: Vespa application instance.
+        """
+        if environment == "dev":
+            region = self.get_dev_region()
+        elif environment == "prod":
+            region = self.get_prod_region()
+        else:
+            raise ValueError("Environment must be 'dev' or 'prod'.")
 
         mtls_endpoint = self.get_mtls_endpoint(
-            instance=instance, region=region, environment="prod"
+            instance=instance, region=region, environment=environment
         )
         if self.auth_client_token_id is not None:
             try:  # May have client_token_id set but the deployed app was not configured to use it
                 token_endpoint = self.get_token_endpoint(
-                    instance=instance, region=region, environment="prod"
+                    instance=instance, region=region, environment=environment
                 )
             except Exception as _:
                 token_endpoint = None
         else:
             token_endpoint = None
-
+        print(f"Connecting to {token_endpoint or mtls_endpoint}", file=self.output)
         app = Vespa(
             url=token_endpoint or mtls_endpoint,
             cert=self.data_cert_path,
-            key=self.data_key_path or None,
+            key=self.data_key_path,
             application_package=self.application_package,
             vespa_cloud_secret_token=os.environ.get(VESPA_CLOUD_SECRET_TOKEN),
         )
-        app.wait_for_application_up(max_wait=max_wait)
-        print("Finished deployment.", file=self.output)
         return app
 
+    def get_deployment_jobs(self) -> List[str]:
+        prod_regions = self.get_prod_regions()
+        return ["system-test", "staging-test"] + [
```

Comment:
This breaks down for more complex deployment specs, but those are perhaps not modelled in `pyvespa` anyway? Depending on what this is used for, it could also be gotten from the API.

Reply:
Deleted that function, as it was only used by the earlier approach!